Tests/AsyncHTTPClientTests/RequestBagTests.swift

```diff
@@ -19,31 +19,32 @@ import NIOEmbedded
 import NIOHTTP1
 import NIOPosix
 import XCTest
+import Atomics
 
 final class RequestBagTests: XCTestCase {
     func testWriteBackpressureWorks() {
         let embeddedEventLoop = EmbeddedEventLoop()
         defer { XCTAssertNoThrow(try embeddedEventLoop.syncShutdownGracefully()) }
         let logger = Logger(label: "test")
 
-        var writtenBytes = 0
-        var writes = 0
-        let bytesToSent = (3000...10000).randomElement()!
-        let expectedWrites = bytesToSent / 100 + ((bytesToSent % 100 > 0) ? 1 : 0)
-        var streamIsAllowedToWrite = false
+        let writtenBytes: ManagedAtomic<Int> = .init(0)
+        let writes: ManagedAtomic<Int> = .init(0)
+        let bytesToSent: ManagedAtomic<Int> = .init((3000...10000).randomElement()!)
```

Comment:
`bytesToSend` doesn't look mutable here. Why do we need an atomic here?

Reply:
I don't think we should be using atomics on any of this, it's a bit too overly clever. Locks are just fine.